### DISCUSSION TITLE
[Bug]: Preview Of Document File is Not Visible In PIM  DAM UI

### DISCUSSION THREAD

**Discussion by praveenjha2112** [pimcore/pimcore#13380] [Q&A / Support]
### Expected behavior

![image](https://user-images.githubusercontent.com/112381479/195753285-7508eda8-a8cc-4588-85b2-427b25430265.png)

### Actual behavior

![image](https://user-images.githubusercontent.com/112381479/195753375-49f8f9e6-ce04-4695-b682-abc103f3d1e7.png)

### Steps to reproduce

1. Upload  attached file in asset 
2. open it in ui.
[Sis_Thai_Test_Cases_Oct_6_2022_V1.5.xlsx](https://github.com/pimcore/pimcore/files/9782138/Sis_Thai_Test_Cases_Oct_6_2022_V1.5.xlsx)

**Comment by brusch** [comment]
Seems your maintenance/messenger:consume is not running properly, or LibreOffice dependency is missing.

**Reply by praveenjha2112** [reply]
Hi @brusch 
![image](https://user-images.githubusercontent.com/112381479/195817366-3e33a1da-95f5-4c21-87d7-7970e727c7ce.png)

can you guide/help me

**Reply by kingjia90** [reply]
@praveenjha2112 please look at here
https://pimcore.com/docs/pimcore/current/Development_Documentation/Getting_Started/Installation.html#page_5-Maintenance-Cron-Job

I've tried with the exact file and works fine. BTW it doesn't look a document that should be publicly available 🙈 better delete it and this discussion

**Reply by praveenjha2112** [reply]
Hi @kingjia90 
![image](https://user-images.githubusercontent.com/112381479/195820832-9e68aa01-1539-40bf-99b2-e3f29b9ef176.png)
is there need  to set this command in cron

**Reply by kingjia90** [reply]
The cron job is necessary in production.

Weird, usually it should work straight away, there must be some config overriding this default line 
https://github.com/pimcore/pimcore/blob/4d05802e97f5f60182442c88356d01d3b22d1905/bundles/CoreBundle/Resources/config/pimcore/default.yaml#L36

**Reply by praveenjha2112** [reply]
I have  set up the cron but still its not working
any idea
above mentioned screenshot is of corebundle config

**Reply by mattamon** [reply]
I tried it locally and even though the maintenance isn't running the document preview worked OOTB. 

Although the preview did not work the first time opening it. Directly after uploading the preview won't load. 
On second try it does and then it works everytime.

Does any config in your source code override the corebundle config like @kingjia90 mentioned?

**Reply by praveenjha2112** [reply]
Hi @mattamon 
I have tried it locally and also done all the changes and still is not working

**Reply by dvesh3** [reply]
@praveenjha2112 before you try to setup the `messenger:consume`, please make sure you are using pimcore version >= 10.2 (as the command was introduced in this version). If it is not the case, then please search for older issues related to this problem and then decide to upgrade or apply the patch on current version. thanks!

**Reply by MosiiAUT** [reply]
Same Problem here with a .docx
Pimcore version: 10.4.6
LibreOffice is installed
Cronjobs are all set up but no previews are generated. There are also no messages in the queue

**Reply by drs-anton** [reply]
Hi
Before we open a new Bug we will try it here.

We have the exact same problem. But .pdfs show up. Only .docx documents don't load.
But including that, the thumbnail generation for documents does not work as well. (.pdf and .docx)

Maintenance job as well as the messenger queues (including pimcore_asset_update) are running via supervisor.
No errors are written in logs which could lead to a solution for that particular problems.

Used versions of dependencies on our server (Debian 11.8):
LibreOffice 7.0.4.2 00(Build:2)
Ghostscript 9.53.3
Chromium 119.0.6045.159

**Comment by MosiiAUT** [comment]
Hello, i found a solution for our problem

Commands that are started via cronjob often don't find the path to de executables, which the messenger queue unfortunately is.
This can be fixed fia this line on the top of the Crontab:
PATH=/bin:/usr/bin:/usr/local/bin

At least this fixed it in our project setup, I just tried it.
Don't forget to stop the current workers.

I don't really know if this can be fixed in Pimcore itself. The executable is found and shown in the "System-Requirements Check".
This is just a problem with commands executed via the cron.

**Reply by drs-anton** [reply]
Hi @MosiiAUT 
Thanks for the suggestion.
We will try that as well. But as I mentioned we don't run the maintenance and queue jobs as cronjobs but with supervisor.

The next strange thing to mention is, that on the requieremts UI not everything (including LibreOffice) is marked as ok. But everything is installed as per Pimcore Documentation and the binaries are there. Otherwise I would not be able to provide the installed versions.

<img width="734" alt="image" src="https://github.com/pimcore/pimcore/assets/135015721/54007ca1-39ac-4fa3-9d75-b7092353f1fc">
